commit d4842e99
Author: BenjiReis <benjamin.reis@vates.fr>
Date:   Mon Nov 16 09:37:47 2020 +0100

    add DNS servers when either IPv4 or IPv6 is configured and not using DHCP
    
    Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

diff --git a/networkd/network_server.ml b/networkd/network_server.ml
index 8f2de8df..5f081dfd 100644
--- a/networkd/network_server.ml
+++ b/networkd/network_server.ml
@@ -666,16 +666,16 @@ module Interface = struct
                 ) ) ->
                 update_config name c ;
                 exec (fun () ->
-                    (* We only apply the DNS settings when in static IPv4 mode
-                       to avoid conflicts with DHCP mode. * The `dns` field
+                    (* We only apply the DNS settings when not in a DHCP mode
+                       to avoid conflicts. * The `dns` field
                        should really be an option type so that we don't have to
                        derive the intention * of the caller by looking at other
                        fields. *)
-                    match ipv4_conf with
-                    | Static4 _ ->
-                        set_dns () dbg ~name ~nameservers ~domains
-                    | _ ->
-                        ()) ;
+                    match (ipv4_conf, ipv6_conf) with
+                    | (Static4 _, _)
+                    | (_, Static6 _)
+                    | (_, Autoconf6) -> set_dns () dbg ~name ~nameservers ~domains
+                    | _ -> ());
                 exec (fun () -> set_ipv4_conf dbg name ipv4_conf) ;
                 exec (fun () ->
                     match ipv4_gateway with
